client/app/src/components/account/votes-tab/votes-tab.controller.js

```diff
@@ -0,0 +1,97 @@
+;(function () {
+  'use strict'
+
+  let VotesTabController = function VotesTabController ($scope, $mdDialog, accountService, networkService, toastService, TRANSACTION_TYPES) {
+    this.accountAddress = ''
+    this.delegates = []
+    this.network = networkService.getNetwork()
+    this.voteModal = {}
+    this.ul = {}
+
+    this.$onInit = () => {
+      this.ul = this.accountCtrl // TODO depricate
+    }
+
+    this.getDelegateList = (accountObj) => {
+      let delegateList = []
+
+      if (accountObj.selectedVotes) {
+        delegateList = accountObj.selectedVotes.filter((item, index) => {
+          return accountObj.selectedVotes.map(mapObj => mapObj.username).indexOf(item.username) === index
+        })
+      }
+
+      return delegateList
+    }
+
+    this.vote = (accountObj, delegateToUnvote) => {
+      this.voteModal = $mdDialog.show({
+        templateUrl: './src/components/account/votes-tab/templates/vote.dialog.html',
+        controller: 'VoteDialogController',
+        controllerAs: '$dialog',
+        clickOutsideToClose: false,
+        resolve: {
+          accountObj: () => accountObj,
+          delegateToUnvote: () => delegateToUnvote,
+          passphrasesArr: () => accountService.getPassphrases(this.account.address),
+          activeDelegates: () => {
+            return accountService
+              .getActiveDelegates()
+              .then(delegates => delegates)
+              .catch(() => {
+                toastService.error('Could not fetch active delegates - please check your internet connection')
+              })
+          },
+          currentTheme: () => this.theme
+        }
+      })
+
+      this.voteModal.then(payload => {
+        if (payload.new_delegate) {
+          let voteActionChar = delegateToUnvote ? '-' : '+'
+          let transactionObj = {
+            ledger: this.account.ledger,
+            publicKey: this.account.publicKey,
+            fromAddress: this.account.address,
+            publicKeys: `${voteActionChar}${payload.new_delegate.publicKey}`,
+            masterpassphrase: payload.passphrases.first,
+            secondpassphrase: payload.passphrases.second
+          }
+
+          accountService.createTransaction(TRANSACTION_TYPES.VOTE, transactionObj).then((transaction) => {
```

Comment:
I think you need to use the `transactionBuilderService` instead of the `accountService`: `transactionBuilderService.createVoteTransaction(transactionObj)`

Reply:
Great catch @luciorubeens, this should be working in my latest push.